### PR TITLE
core/libutee: perform cleanup for magic "4"

### PR DESCRIPTION
**Pull request by igoropaniuk**
Perform cleanup for magic "4" constant that represents amount of tee
params

`Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

### CODE REVIEW

**Comment by etienne-lms** [comment]
Looks ok but checkpatch complains to fix.

**Comment by igoropaniuk** [comment]
@etienne-lms fixed

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: lib/libutee/include/tee_ta_api.h [hunk 1]
```diff
@@ -192,9 +192,9 @@ void TA_EXPORT TA_CloseSessionEntryPoint(void *sessionContext);
  */
 
 TEE_Result TA_EXPORT TA_InvokeCommandEntryPoint(void *sessionContext,
-						uint32_t commandID,
-						uint32_t paramTypes,
-						TEE_Param params[4]);
+			uint32_t commandID,
+			uint32_t paramTypes,
+			TEE_Param params[TEE_NUM_PARAMS]);
```
Comment (jforissier) [inline lib/libutee/include/tee_ta_api.h]: Why change the indentation?

File: lib/libutee/tee_api.c [hunk 2]
```diff
@@ -99,7 +99,8 @@ void __utee_to_param(TEE_Param params[TEE_NUM_PARAMS],
 
 TEE_Result TEE_OpenTASession(const TEE_UUID *destination,
 			     uint32_t cancellationRequestTimeout,
-			     uint32_t paramTypes, TEE_Param params[4],
+			     uint32_t paramTypes,
+				 TEE_Param params[TEE_NUM_PARAMS],
```
Comment (jforissier) [inline lib/libutee/tee_api.c]: Bad indentation

File: lib/libutee/tee_api.c [hunk 3]
```diff
@@ -136,7 +137,8 @@ void TEE_CloseTASession(TEE_TASessionHandle session)
 TEE_Result TEE_InvokeTACommand(TEE_TASessionHandle session,
 			       uint32_t cancellationRequestTimeout,
 			       uint32_t commandID, uint32_t paramTypes,
-			       TEE_Param params[4], uint32_t *returnOrigin)
+				   TEE_Param params[TEE_NUM_PARAMS],
+				   uint32_t *returnOrigin)
```
Comment (jforissier) [inline lib/libutee/tee_api.c]: Bad indentation

File: lib/libutee/include/tee_api.h [hunk 4]
```diff
@@ -88,16 +88,18 @@ void TEE_Panic(TEE_Result panicCode);
 
 TEE_Result TEE_OpenTASession(const TEE_UUID *destination,
 			     uint32_t cancellationRequestTimeout,
-			     uint32_t paramTypes, TEE_Param params[4],
+				 uint32_t paramTypes,
+				 TEE_Param params[TEE_NUM_PARAMS],
```
Comment (jforissier) [inline lib/libutee/include/tee_api.h]: Bad indentation

File: lib/libutee/include/tee_api.h [hunk 5]
```diff
@@ -88,16 +88,18 @@ void TEE_Panic(TEE_Result panicCode);
...
 
 void TEE_CloseTASession(TEE_TASessionHandle session);
 
 TEE_Result TEE_InvokeTACommand(TEE_TASessionHandle session,
 			       uint32_t cancellationRequestTimeout,
-			       uint32_t commandID, uint32_t paramTypes,
-			       TEE_Param params[4], uint32_t *returnOrigin);
+				   uint32_t commandID, uint32_t paramTypes,
+				   TEE_Param params[TEE_NUM_PARAMS],
+				   uint32_t *returnOrigin);
```
Comment (jforissier) [inline lib/libutee/include/tee_api.h]: Bad indentation

File: core/arch/arm/include/kernel/static_ta.h [hunk 6]
```diff
@@ -40,11 +40,12 @@ struct static_ta_head {
 	TEE_Result (*create_entry_point)(void);
 	void (*destroy_entry_point)(void);
 	TEE_Result (*open_session_entry_point)(uint32_t nParamTypes,
-			TEE_Param pParams[4], void **ppSessionContext);
+					TEE_Param pParams[TEE_NUM_PARAMS],
+					void **ppSessionContext);
```
Comment (jforissier) [inline core/arch/arm/include/kernel/static_ta.h]: For consistency, use the same indentation as `invoke_command_entry_point` below.

File: lib/libutee/include/tee_ta_api.h (hunk 1 above)
Comment (igoropaniuk) [inline lib/libutee/include/tee_ta_api.h]: because checkpatch fails

Comment (jforissier) [inline lib/libutee/include/tee_ta_api.h]: Ah, right. I would probably added one more tab still, but it's OK like this ;)

**Comment by igoropaniuk** [comment]
are some additional changes needed?

**Comment by jforissier** [comment]
Fine by me.
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by igoropaniuk** [comment]
Applied tags and rebased on master

**Pull request merged**